File: test/fixtures/node/multipart_post.js
import fetch, { FormData, fileFromSync } from 'node-fetch';

const form = new FormData();
form.append('attributes', '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}');
form.append('file', fileFromSync('myfile.jpg'));

fetch('http://localhost:28139/api/2.0/files/content', {
  method: 'POST',
  headers: {
    'Authorization': 'Bearer ACCESS_TOKEN'
  },
  body: form
});

File: test/fixtures/r/multipart_post.r
library(httr)

headers = c(
  Authorization = "Bearer ACCESS_TOKEN"
)

files = list(
  attributes = '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}',
  file = upload_file("myfile.jpg")
)

res <- httr::POST(url = "http://localhost:28139/api/2.0/files/content", httr::add_headers(.headers=headers), body = files, encode = "multipart")
